src/zenml/cli/hub.py

```diff
@@ -0,0 +1,1203 @@
+
+        # Check if plugin can be uninstalled
+        package_name = plugin.package_name
+        if not package_name:
+            error(
+                f"Plugin '{display_name}' is not available for uninstallation."
+            )
+
+        # pip uninstall the wheel
+        logger.info(f"Uninstalling plugin '{display_name}'...")
+        subprocess.check_call(
+            [sys.executable, "-m", "pip", "uninstall", package_name, "-y"]
+        )
+        logger.info(f"Successfully uninstalled plugin '{display_name}'.")
+
+
+@hub.command("clone")
+@click.argument("plugin_name", type=str, required=True)
+@click.option(
+    "--version",
+    "-v",
+    type=str,
+    help="Version of the plugin to clone.",
+)
+@click.option(
+    "--output_dir",
+    type=str,
+    help="Output directory to clone the plugin to.",
+)
+def clone_plugin(
+    plugin_name: str,
+    version: Optional[str] = None,
+    output_dir: Optional[str] = None,
+) -> None:
+    """Clone a plugin from the hub.
+
+    Args:
+        plugin_name: Name of the plugin.
+        version: Version of the plugin. If not specified, the latest version
+            will be used.
+        output_dir: Output directory to clone the plugin to. If not specified,
+            the plugin will be cloned to a directory with the same name as the
+            plugin in the current working directory.
+    """
+    with event_handler(
+        event=AnalyticsEvent.ZENML_HUB_PLUGIN_CLONE,
+        metadata={
+            "plugin_name": plugin_name,
+            "plugin_version": version,
+        },
+    ) as analytics_handler:
+        client = HubClient()
+        analytics_handler.metadata["hub_url"] = client.url
+        plugin_name, author = _parse_plugin_name(plugin_name)
+        analytics_handler.metadata["plugin_name"] = plugin_name
+        analytics_handler.metadata["plugin_author"] = author
+        display_name = _plugin_display_name(plugin_name, version, author)
+
+        # Get plugin from hub
+        plugin = client.get_plugin(
+            plugin_name=plugin_name, plugin_version=version, author=author
+        )
+        if not plugin:
+            error(f"Could not find plugin '{display_name}' on the hub.")
+        analytics_handler.metadata["plugin_version"] = plugin.version
+
+        repo_url = plugin.repository_url
+        subdir = plugin.repository_subdirectory
+        commit = plugin.repository_commit
+
+        # Clone the source repo
+        if output_dir is None:
+            output_dir = os.path.join(os.getcwd(), plugin_name)
+        logger.info(f"Cloning plugin '{display_name}' to {output_dir}...")
+        # If no subdir is set, we can clone directly into output_dir
+        if not subdir:
+            repo_path = plugin_dir = output_dir
+        # Otherwise, we need to clone into a random dir and then move the subdir
+        else:
+            random_repo_name = f"_{uuid4()}"
+            repo_path = os.path.abspath(
```

Comment:
This seems like a tempdir to me, not a random string in the current working directory

Reply:
yes it is, how would you like it to be changed then? Are you suggesting I use some existing temp dir utils we have or should I simply change the variable name?

Reply:
```python
import tempfile

with tempfile.TemporaryDirectory(...):
  ...
```

Reply:
While not likely, your current approach could still crash if it doesn't get to the finally clause (which might just be a `CTRL-C` from the user at an unfortunate moment.

Reply:
ah cool, thanks, changed it accordingly 👍